api/Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM php:7.4.27-apache-bullseye as base
+
+# Install pdo_pgsql extension.
+RUN apt-get update \
+    && apt-get install -y libpq-dev \
+    && docker-php-ext-install pdo_pgsql
+
+WORKDIR /var/www/html/api
+
+COPY .env.example .env
+COPY composer.json composer.lock ./
+COPY tests/ tests/
+
+FROM base as builder
+# Install zip ext required for composer install from dist.
+RUN apt-get update \
+    && apt-get install -y libzip-dev \
+    && docker-php-ext-install zip
+# Copy composer binary from official image.
+COPY --from=composer:latest /usr/bin/composer /usr/local/bin/composer
```

Comment:
Is using "latest" here alright/consistent? Dependabot won't help us update docker image tags here, so I figured easiest not to lock it.

Reply:
Oh, this is very clever, using it from the image rather than downloading a binary into the container. 🤯  I'd suggest locking it to version 2 so we don't get a surprise bump to 3.0.0 whenever that happens.

Reply:
```suggestion
COPY --from=composer:2 /usr/bin/composer /usr/local/bin/composer
```